app_check/integration_test/src/integration_test.cc

```diff
@@ -335,6 +337,31 @@ TEST_F(FirebaseAppCheckTest, TestSignIn) {
   EXPECT_NE(auth_->current_user(), nullptr);
 }
 
+TEST_F(FirebaseAppCheckTest, TestDebugProviderValidToken) {
+  firebase::app_check::DebugAppCheckProviderFactory* factory =
+      firebase::app_check::DebugAppCheckProviderFactory::GetInstance();
+#if FIREBASE_PLATFORM_IOS
+  ASSERT_NE(factory, nullptr);
+  InitializeApp();
+  firebase::app_check::AppCheckProvider* provider =
+      factory->CreateProvider(app_);
+  ASSERT_NE(provider, nullptr);
+  auto token_callback{[](firebase::app_check::AppCheckToken token,
+                         int error_code, const std::string& error_message) {
+    EXPECT_EQ(firebase::app_check::kAppCheckErrorNone, error_code);
+    EXPECT_EQ("", error_message);
+    EXPECT_NE(0, token.expire_time_millis);
+    EXPECT_NE("", token.token);
+  }};
+  provider->GetToken(token_callback);
+  // Sleep for a bit to give GetToken time to succeed
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Using a semaphore, or even just a spin loop that checks some variable the callback can change seems like it would be better to guarantee the callback actually happens.

Reply:
I tried <semaphore.h> but found that it was not supported on iOS. I also noticed that app/src/semaphore.h has different implementations per platform, and so getting a semaphore to work in a test seems like extra work.
I'll try to get a std::future or spin loop or something similar to work.

Reply:
I updated the test to use std::promise / std::future in order to synchronize.